server/src/main/java/org/opensearch/indices/recovery/MultiFileWriter.java

```diff
@@ -128,6 +131,22 @@ public IndexOutput openAndPutIndexOutput(String fileName, StoreFileMetadata meta
         return indexOutput;
     }
 
+    /**
+     * Helper method that determines if we should fsync temporary file being copied by MultiFileWriter. In the case of segment replication,
+     * we only want to fsync files that are part of a greater commit point while we fsync all files in the case of document replication.
+     * @return boolean - true if we should fsync temporary file; false if not
+     */
+    private boolean shouldFsync() {
+        if (store.indexSettings().isSegRepEnabled()) {
+            List<String> filenames = indexState.fileDetails()
```

Comment:
This can be a single stream.

```java
                return indexState.fileDetails()
                .stream()
                .map(ReplicationLuceneIndex.FileMetadata::name)
                .anyMatch((filename) -> filename.startsWith(IndexFileNames.SEGMENTS));
```